### ISSUE TITLE
CleanLearning default classifier

### ISSUE THREAD

**Issue by jwmueller** [cleanlab/cleanlab#406]
Currently CleanLearning can be run like this which is nice for users who don't know what classifier to use for their data:
```
cl = CleanLearning()
cl.find_label_issues(data, labels)
```
but it always defaults to sklearn's LogisticRegression, which may not work for many types of `data`. Consider deferring the choice of default classifier until `data` is provided and then selecting from a broader suite of default options beyond LogisticRegression as well to make this work for more data types.  Challenge is how to do this without introducing additional dependencies for the cleanlab package, or making this code too complex.

**Comment by jwmueller** [comment]
Yes totally! But can you please articulate a clear plan on how you'd like to go about this? We want to be careful about introducing more dependencies for this package, so preferably any new code will stay within sklearn ecosystem.

If it helps to have real-time interactive discussion, feel free to message me in Slack:  https://cleanlab.ai/slack

**Comment by milmin** [comment]
Hello, is anybody already working on this? I'd like to contribute if possible

**Comment by jwmueller** [comment]
> Hello, is anybody already working on this? I'd like to contribute if possible

Awesome @milmin !  If it helps to have real-time interactive discussion, feel free to message me in Slack: https://cleanlab.ai/slack

What do you think about working on this alternate (closely related) Datalab issue instead?
https://github.com/cleanlab/cleanlab/issues/735

Because I believe @Steven-Yiran is already working on the current CleanLearning issue, feel free to update me otherwise.

**Comment by Steven-Yiran** [comment]
Yes, I am still working on the flatten multi-dimension input part. Should have result this weekend.
